File: javascript/decorator/StarbuzzCoffee/decorator/condiments/Whip.js
import {CondimentDecorator} from "../CondimentDecorator.js";
import {Beverage} from "../../component/Beverage.js";

export class Whip extends CondimentDecorator {

    beverage = new Beverage()

    constructor(beverage) {
        super();
        this.beverage = beverage
    }

    getDescription() {
        return this.beverage.getDescription() + ", Whip"
    }

    cost() {
        return this.beverage.cost() + 0.11
    }

}

File: cpp/decorator/StarbuzzCoffee/decorator/condiments/Whip.cpp
#include "../CondimentDecorator.h"

class Whip : public CondimentDecorator {

public:
    explicit Whip(Beverage* beverage) { this->beverage = beverage; }

    string getDescription() override { return beverage->getDescription() + ", Whip"; }

    double cost() override { return beverage->cost() + 0.10; }

private:
    Beverage* beverage;

};

